ui/src/message_popup/author/question_page.jsx

```diff
@@ -0,0 +1,284 @@
+/* @flow weak */
+/* eslint no-console: "off" */
+import React from 'react';
+import _ from 'lodash';
+import VelocityTransitionGroup from 'velocity-react/velocity-transition-group';
+import 'velocity-animate/velocity.ui';
+
+import * as EditingComponents from './question_editing_components/editing_component.jsx';
+
+import {allQuestions} from '../questions.js';
+import {allStudents} from '../../data/virtual_school.js';
+import {learningObjectives} from '../../data/learning_objectives.js';
+import {indicators} from '../../data/indicators.js';
+import * as Routes from '../../routes.js';
+
+import AppBar from 'material-ui/AppBar';
+import IconButton from 'material-ui/IconButton';
+import RaisedButton from 'material-ui/RaisedButton';
+import Dialog from 'material-ui/Dialog';
+import FlatButton from 'material-ui/FlatButton';
+
+import ArrowBackIcon from 'material-ui/svg-icons/navigation/arrow-back';
+
+export default React.createClass({
+  displayName: 'QuestionPage',
+
+  propTypes: {
+    originalQuestion: React.PropTypes.object,
+    questionText: React.PropTypes.string,
+    students: React.PropTypes.array,
+    learningObjective: React.PropTypes.object,
+    indicator: React.PropTypes.object,
+    goodExamplesText: React.PropTypes.string,
+    badExamplesText: React.PropTypes.string
+  },
+
+  getDefaultProps(){
+    return ({
+      questionText: '',
+      students: [],
+      learningObjective: learningObjectives[0],
+      indicator: indicators[0],
+      goodExamplesText: "",
+      badExamplesText: ""
+    });
+
+  },
+  getInitialState() {
+    return ({
+      questionText: this.props.questionText,
+      students: this.props.students,
+      learningObjective: this.props.learningObjective,
+      indicator: this.props.indicator,
+      goodExamplesText: this.props.goodExamplesText,
+      badExamplesText: this.props.badExamplesText,
+      deleteConfirmationOpen: false
+    });
+  },
+
+  returnToQuestions(){
+    Routes.navigate(Routes.messagePopupAuthorQuestionsPath());
+  },
+
+  onQuestionTextChange(event, value){
+    this.setState({questionText: value});
+  },
+
+  addStudent(studentName){
+    const student = _.find(allStudents, student => student.name.toLowerCase() === studentName.toLowerCase());
+    const alreadyAdded =  student !== undefined && _.find(this.state.students, newStudent => newStudent.id === student.id) !== undefined;
+    if(student !== undefined && !alreadyAdded){
```

Comment:
Would it work to filter the list of students so that students already added couldn't be added again, rather than having to guard for that here?